src/gg.regression.unity.bots/Runtime/Scripts/StateActionTypes/RGStateBehaviour.cs

```diff
@@ -0,0 +1,44 @@
+using UnityEngine;
+
+namespace RegressionGames.StateActionTypes
+{
+    public interface IRGStateBehaviour
+    {
+        public void PopulateStateEntity(RGStateEntity_Core gameObjectCoreState, out bool isPlayer);
+    }
+    
+    public abstract class RGStateBehaviour<T> : MonoBehaviour , IRGStateBehaviour where T : IRGStateEntity
+    {
+        // cache this so we don't create every tick
```

Comment:
Yeah at some point we should do some official performance testing just to keep track

Reply:
That actually might work well as a test - have a CI/CD pipeline that can compare timing between our SDK and not using the SDK. Def too much work for now but would be cool